spec/requests/api/v1/driver_commission_histories_spec.rb

```diff
@@ -1,24 +1,90 @@
 require 'rails_helper'
 
-RSpec.describe '/api/v1/all', :type => :request do
-  let(:all) do
-    DriverCommissionHistoryAdapter.all
-  end
+RSpec.describe '/api/v1/driver_commission_history', :type => :request do
+  describe 'all' do
+    let(:all) do
+      DriverCommissionHistoryAdapter.all
+    end
 
-  let(:by_paid_date) do
-    DriverCommissionHistoryAdapter.by_paid_date(Date.new(2012, 01, 01))
-  end
+    let(:all_url) { '/api/v1/driver_commissions_history' }
+
+    context 'when application id and secret are valid' do
+      let(:response) { doorkeeper_request.get(all_url) }
+      # it { expect(doorkeeper_request.token).to eq('')}
+      it 'responds with a body equal to json' do
+        expect(response.body).to eq(all.to_json)
+      end
+    end
 
-  it 'responds with all driver commission history json' do
-    get '/api/v1/driver_commissions_history'
+    context 'when application id and secret are not valid' do
```

Comment:
I think this looks a little funky and think it would read better if it used the doorkeeper_request that you're using above... I would actually considering changing the doorkeeper_request to some kind of doorkeeper_get that takes a url, app id and secret and sets the response (at an instance level?). I think that may allow us to treat it in a similar fashion as the usual controller spec get method.

I'm happy to discuss my thoughts further if you like.